### PR TITLE
Improve the glossary.

### PR DESCRIPTION
**Pull request by varsill**

### CODE REVIEW

File: glossary/glossary.md [hunk 1]
```diff
@@ -1,87 +1,155 @@
...
+    - #### **P frame**
+    A P frame (Predictive frame) is a video compression frame that depends on the preceding I frame or P frame for data, using the motion differences between the two frames to reduce the amount of stored information and make the file size smaller.
+    - #### **B frame**
+    A B frame (Bidirectional predictive frame) is a video compression frame that relies on both the preceding and following I or P frames for data, using the motion differences between these frames to further reduce the stored information, resulting in even smaller file sizes and better compression efficiency.
 ## Membrane Framework
-
+- #### **Action**
+  An action can be returned from [callback][#callback] and it is a way of element interaction with other elements and parts of the framework. An exemplary action might be [`:buffer`](https://hexdocs.pm/membrane_core/Membrane.Element.Action.html#t:buffer/0) action, that sends buffers through a pad, or [`:terminate`](https://hexdocs.pm/membrane_core/Membrane.Element.Action.html#t:terminate/0) action, that terminates element with given reason.
```
Comment (jerzywilczek) [inline glossary/glossary.md:122]:
```suggestion
  An action can be returned from [callback](#callback) and it is a way of element interaction with other elements and parts of the framework. An exemplary action might be [`:buffer`](https://hexdocs.pm/membrane_core/Membrane.Element.Action.html#t:buffer/0) action, that sends buffers through a pad, or [`:terminate`](https://hexdocs.pm/membrane_core/Membrane.Element.Action.html#t:terminate/0) action, that terminates element with given reason.
```

File: glossary/glossary.md [hunk 2]
```diff
@@ -1,87 +1,155 @@
...
+- #### **Callback**
+  A callback is a function defined by a user, that gets called once a particular event happens.
+  Exemplary callbacks in the Membrane Framework are: [`handle_end_of_stream/3`](https://hexdocs.pm/membrane_core/Membrane.Element.WithInputPads.html#c:handle_end_of_stream/3) being called once the end of stream event is received on some pad or []`handle_init/2`](https://hexdocs.pm/membrane_core/Membrane.Pipeline.html#c:handle_init/2) called on initialization of an [element](#element).
```
Comment (jerzywilczek) [inline glossary/glossary.md:125]:
```suggestion
  Exemplary callbacks in the Membrane Framework are: [`handle_end_of_stream/3`](https://hexdocs.pm/membrane_core/Membrane.Element.WithInputPads.html#c:handle_end_of_stream/3) being called once the end of stream event is received on some pad or [`handle_init/2`](https://hexdocs.pm/membrane_core/Membrane.Pipeline.html#c:handle_init/2) called on initialization of an [element](#element).
```

File: glossary/glossary.md [hunk 3]
```diff
@@ -1,87 +1,155 @@
...
   - #### **Filter**
-    An element with both input and output pads, which is responsible for transforming data.
+    An element with both input and output [pads](#pads), which is responsible for transforming data.
   - #### **Sink** 
     An element with only input [pads](#pads), the last element of a pipeline. It might be responsible, i.e. for writing the output to the file or playing the incoming media stream.
+  - #### **Endpoint**
+    An element with both input and output [pads](#pads), responsible for receiving and consuming data (e.g., writing to a soundcard, sending via TCP, etc.) as well as producing data (e.g., reading from a soundcard, downloading via HTTP, etc.) and sending it through the corresponding pads. It can be thought of as an element merging all the functionalities of previously mentioned element categories: source, filter, and sink.
+- #### **ChildrenSpec**
+  In Membrane Framework, we call ChildrenSpec a bunch of functionalities that allow for defining the topology of the [pipeline](#pipeline) or a [bin](#bin). You can read more about ChildrenSpec [here](https://hexdocs.pm/membrane_core/Membrane.ChildrenSpec.html).
```
Comment (jerzywilczek) [inline glossary/glossary.md:148]:
```suggestion
  In Membrane Framework, `ChildrenSpec` is a way of describing the topology of a [pipeline](#pipeline) or a [bin](#bin). You can read more about `ChildrenSpec` [here](https://hexdocs.pm/membrane_core/Membrane.ChildrenSpec.html).
```

**Review by jerzywilczek — CHANGES_REQUESTED** [review]
:cowboy_hat_face:

File: glossary/glossary.md [hunk 4]
```diff
@@ -91,16 +159,23 @@
...
+    In Membrane it's an element's action that lets the programmer handle just one buffer at a time. When redemanding, the `handle_demand/5` callback is synchronously called. You can read more about redemands [here](https://hexdocs.pm/membrane_core/Membrane.Element.Action.html#t:redemand/0).
 
 ## General Elixir/Erlang concepts
 - ### OTP Behavior
   - #### **GenServer**
-    Elixir bahaviour abstracts client/server interaction. [https://elixir-lang.org/getting-started/mix-otp/genserver.html](https://elixir-lang.org/getting-started/mix-otp/genserver.html)
-  - #### **Phoenix** 
-    The web development framework written in Elixir. [https://phoenixframework.org/](https://phoenixframework.org/)
-  - #### **Mix**
+    Elixir behavior abstracts client/server interaction. [https://elixir-lang.org/getting-started/mix-otp/genserver.html](https://elixir-lang.org/getting-started/mix-otp/genserver.html)
```
[the remark this answers is not in the document]
Comment (varsill) [reply glossary/glossary.md:177]: Indeed, we can, I have changed it

File: glossary/glossary.md [hunk 5]
```diff
@@ -1,87 +1,155 @@
...
 
   - #### **Packet**
-    It is a formatted unit of data transmitted over network. In order to send data over network it has to be fragmented into packets, which size is limited by [MTU(Maximum Transfer Unit)](https://en.wikipedia.org/wiki/Maximum_transmission_unit) - 1500 bytes when using [Ethernet](https://en.wikipedia.org/wiki/Ethernet_frame).
+    It is a formatted unit of data transmitted over the network. To send data over the network it has to be fragmented into packets, which size is limited by [MTU(Maximum Transfer Unit)](https://en.wikipedia.org/wiki/Maximum_transmission_unit) - 1500 bytes when using [Ethernet](https://en.wikipedia.org/wiki/Ethernet_frame).
   - #### **Frame**
-    Frame can refer to either [network frame](<https://en.wikipedia.org/wiki/Frame_(networking)>) or **media frame**, which is a basic data unit used by media coding formats. In particular one media frame can represent a single image in a video.
+    The frame can refer to either [network frame](<https://en.wikipedia.org/wiki/Frame_(networking)>) or **media frame**, which is a basic data unit used by media coding formats. In particular, one media frame can represent a single image in a video.
```
[the remark this answers is not in the document]
Comment (jerzywilczek) [reply glossary/glossary.md:6]: I think it's fine to leave it as-is in the rest of the definitions

File: glossary/glossary.md [hunk 6]
```diff
@@ -1,87 +1,157 @@
...
 - #### **YUV**
-  A color encoding system that defines one [luminance](https://en.wikipedia.org/wiki/Luminance) and two [chrominance](https://en.wikipedia.org/wiki/Chrominance) components. By reducing the resolution of the chrominance components it is possible to compress an image with minuscule effect on human perception of the image.
+  Color space that defines one [luminance](https://en.wikipedia.org/wiki/Luminance) and two [chrominance](https://en.wikipedia.org/wiki/Chrominance) components. By reducing the resolution of the chrominance components it is possible to compress an image with a minuscule effect on human perception of the image. That encoding is commonly used for analog video processing.
+- #### **YCbCr**
+  Color space that defines one [luminance](https://en.wikipedia.org/wiki/Luminance) and two [chrominance difference] components: [blue-difference](https://en.wikipedia.org/wiki/B-Y) and [red-difference](https://en.wikipedia.org/wiki/R-Y). In contrast to **YUV** it's more often used in digital video processing. It happens that YCbCr is mistakenly called [YUV](#yuv).
```
Comment (jerzywilczek) [inline glossary/glossary.md:59]:
```suggestion
  Color space that defines one [luminance](https://en.wikipedia.org/wiki/Luminance) and two [chrominance difference] components: [blue-difference](https://en.wikipedia.org/wiki/B-Y) and [red-difference](https://en.wikipedia.org/wiki/R-Y). In contrast to **YUV** it's more often used in digital video processing. It happens that YCbCr is often mistakenly called [YUV](#yuv).
```

Comment (jerzywilczek) [inline glossary/glossary.md:59]: You have to link to something if you write 'chrominance difference' in square brackets :wink:

**Review by jerzywilczek — CHANGES_REQUESTED** [review]
:cowboy_hat_face:

File: glossary/glossary.md (hunk 6 above)
Comment (varsill) [reply glossary/glossary.md:59]: done

Comment (varsill) [reply glossary/glossary.md:59]: indeed, done ;)

**Review by jerzywilczek — APPROVED** [review]
:octopus:

**Pull request merged**